### ISSUE TITLE
Deleting a word at the end of line might delete the EOL

### ISSUE THREAD

**Issue by Zzzen** [VSCodeVim/Vim#967]
Please _thumbs-up_ 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

---
### What did you do?
1. create a js file:

```
function () {
  if (true) {
    var x
  }
}
```
1. Enter Normal mode, move cursor to x
2. Input `d a w`

```
function () {
  if (true) {
    var }
}
```
### What did you expect to happen?

The EOL should not be removed.
### What happened instead?

The EOL is removed.
### Technical details:

This bug only effects nested blocks.
- VSCode Version: 1.6.1
- VsCodeVim Version: 0.3.8
- OS: win10

**Comment by rebornix** [comment]
This bug comes again, I debt we didn't add test cases for it last time when we fixed it.

**Comment by johnfn** [comment]
> This bug only effects nested blocks.

😕

**Comment by Chillee** [comment]
This issue is actually the same as this issue #1350, and should have been fixed in PR #1549 .

It's not that the bug only affects nested blocks, but that the bug occurs when the word on the next line has whitespace before it.

This bug shouldn't occur in VSCodeVim 0.6.18.

**Comment by xconverge** [comment]
You are right just tested on this version and previous version!
